### DISCUSSION TITLE
API KEY passed through parameter on initialization instead of ENV

### DISCUSSION THREAD

**Discussion by gotkd21** [langchain-ai/langsmith-sdk#835] [Q&A]
### Feature request

Suggest also making it an option to include variable on the initialization call 'client(....., langchain_api_key = xxx).

Same paradigm used by the model providers to send requests programmatically.

### Motivation

I'm thinking of using Langsmith with an application running in AWS Lambda.  Exposing an API_KEY directly as an ENV variable is insecure, and quite frankly, AWS makes it difficult to do.  This change would also help me store all of my secrets and consume them in the exact same way....  as this is the ONLY API_KEY that I use thats required to be stored in the environment.

**Comment by hinthornw** [comment]
Hi @gotkd21 - this is already available. Could you elaborate on what is hindering you?

**Comment by gotkd21** [comment]
thanks... can you point me to the doc on that?  Guess I just couldn't find it anywhere... everything I found just says use the environment variable.

**Comment by hinthornw** [comment]
https://docs.smith.langchain.com/how_to_guides/tracing/trace_with_langchain#trace-without-setting-environment-variables

For if you're using langcahin.

If you're using langsmith it's similar: pass in the Client manually via `langsmith_extra`

**Comment by hinthornw** [comment]
Gonna convert this to a discussion!
